internal/rpc/block_read_stream.go

```diff
@@ -101,8 +101,8 @@ func (s *blockReadStream) Read(b []byte) (int, error) {
 	for i := 0; i < chunksToRead; i++ {
 		chunkOff := i * s.chunkSize
 		chunkEnd := chunkOff + s.chunkSize
-		if chunkEnd >= len(b) {
-			chunkEnd = len(b)
+		if chunkEnd >= n {
```

Comment:
This looks right, but what led you to change it? I should maybe push it upstream separately so we can track if it causes any issues.

Reply:
it was causing a test to fail because the `b` that was used was sized to 512 but the test is the `foo.txt` which only has 4 bytes. As a result the checksum would fail because chunkEnd was the full chunksize rather than only the 4 bytes for the data.

Reply:
Which test? I'm going to see if I can add a failing one to master.

Reply:
I'm pretty sure it was the client_test.go `TestReadFile` i'd have to double check though.

Reply:
Huh, I can't figure out how to reproduce this issue on master. Do you have any idea what it is the wrapped conn is doing that triggers it?

I just tried this:

```
type randomReadConn struct {
	net.Conn
}

func (r *randomReadConn) Read(b []byte) (int, error) {
	return r.Conn.Read(b[0:rand.Intn(len(b)+1)])
}

func TestFileBigReadWeirdSizesMisalignment(t *testing.T) {
	client := getClient(t)
	dial := client.options.DatanodeDialFunc
	if dial == nil {
		dial = (&net.Dialer{}).DialContext
	}

	client.options.DatanodeDialFunc = func(ctx context.Context, network, address string) (net.Conn, error) {
		conn, err := dial(ctx, network, address)
		if err != nil {
			return nil, err
		}

		return &randomReadConn{conn}, nil
	}

	file, err := client.Open("/_test/mobydick.txt")
	require.NoError(t, err)

	hash := crc32.NewIEEE()
	n, err := io.Copy(hash, file)
	assert.NoError(t, err)
	assert.EqualValues(t, n, 1257276)
	assert.EqualValues(t, 0x199d1ae6, hash.Sum32())
}
```